BSP/Simulator/Src/Simulator.c

```diff
@@ -30,6 +30,7 @@ uint8_t stateCount = 0;
 
 // linked list of simulator states
 static simulator_state *states = NULL;
+BPS_OS_MUTEX SimMutex;
```

Comment:
I wouldn't use a BPS RTOS mutex. I think it would make more sense if this was protected with OS calls outside of RTOS stuff. @ray4477 @ClarkPoon @tasniabari might know better.